R/method-correlation.R

```diff
@@ -136,11 +145,29 @@ setMethod("correlation", signature("data.table", "missing"), function(data1, dat
 
   ## Compute correlation
   # rownames and colnames should be the same in this case
-  # na.or.complete removes rows with NAs, if no rows remain then correlation is NA
   # keep matrix for now so we can use lower.tri later, expand.grid will give us the needed data.frame
-  corrResult <- Hmisc::rcorr(as.matrix(data1), type = method)
-  pVals <- corrResult$P
-  corrResult <- corrResult$r
+  if (method == 'sparcc') {
+
+    # get the sparcc function from the SpiecEasi namespace and make a local copy that do.call can find
+    sparcc <- get("sparcc", asNamespace("SpiecEasi"))
+    # sub-sampled `statistic` functions to pass to `boot::boot` that we can use to find pvalues
+    statisticperm=function(data, indices) do.call("sparcc", c(list(apply(data[indices,], 2, sample)), list()))$Cor
+    statisticboot=function(data, indices) do.call("sparcc", c(list(data[indices,,drop=FALSE]), list()))$Cor
+
+    # calling the bootstrap version of sparcc and finding pvalues
+    corrResult <- SpiecEasi::pval.sparccboot(SpiecEasi::sparccboot(data1, statisticboot = statisticboot, statisticperm = statisticperm, R = 100))
```

Comment:
Can we name this something different? It's confusing when variables get redefined later

Reply:
looking below i see we're already in a pattern of doing this lol